src/plugins/intel_gna/transformations/insert_copy_layer.cpp

```diff
@@ -0,0 +1,358 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/insert_copy_layer.hpp"
+#include <openvino/cc/ngraph/itt.hpp>
+#include <ngraph/rt_info.hpp>
+#include <ngraph/opsets/opset8.hpp>
+#include <ngraph/pattern/op/or.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include <transformations/utils/utils.hpp>
+#include <ops/copy.hpp>
+#include <legacy/ngraph_ops/crop_ie.hpp>
+#include <ie/ie_common.h>
+#include <ie_algorithm.hpp>
+#include <openvino/core/except.hpp>
+
+#include "gna_plugin_log.hpp"
+#include "gna_lib_ver_selector.hpp"
+#include "layers/gna_permute.hpp"
+
+namespace GNAPluginNS {
+
+NGRAPH_RTTI_DEFINITION(InsertCopyBeforeMemoryLayer, "InsertCopyBeforeMemoryLayer", 0);
+NGRAPH_RTTI_DEFINITION(InsertCopyBeforeConcatLayer, "InsertCopyBeforeConcatLayer", 0);
+NGRAPH_RTTI_DEFINITION(HandleMultiConnectedLayerToConcatAndMemory, "HandleMultiConnectedLayerToConcatAndMemory", 0);
+NGRAPH_RTTI_DEFINITION(MatchNonComputationalLayers, "MatchNonComputationalLayers", 0);
+NGRAPH_RTTI_DEFINITION(HandleNonComputationalSubgraphs, "HandleNonComputationalSubgraphs", 0);
+
+
+namespace {
+
+    bool is_aligned_split(std::shared_ptr<ngraph::Node> input_op, size_t input_op_out_index) {
+        size_t offset = 0;
+        // currently split layer only supports 2 bytes in int16 and int8 mode. In fp32 mode this is not necessary but is useful for testing
+        const int bytesPerSplitElement = 2;
```

Comment:
IMO we should share and reuse `bytesPerSplitElement` constant which is already defined in:
`void InsertSplitAligningFilterPass::run()`

Reply:
Created common variable